pkg/azuremachine/validate_update.go

```diff
@@ -52,7 +52,14 @@ func (a *UpdateValidator) Validate(ctx context.Context, request *v1beta1.Admissi
 		return microerror.Maskf(errors.ParsingFailedError, "unable to parse AzureMachine CR: %v", err)
 	}
 
-	err := checkSSHKeyIsEmpty(ctx, azureMachineNewCR)
+	err := azureMachineNewCR.ValidateUpdate(azureMachineOldCR)
+	if CAPZIsSSHFieldValidationError(err) {
```

Comment:
Did you check if upstream validation executes all steps and returns collection of errors in the end or if the validation stops on first error?

Could it happen that due to "ignored error" rest of the CAPZ validation chain ends up being silently ignored?

Reply:
Damn, you're right... All validation errors are skipped if there is an SSH field error there. I'll fix it 😬

Reply:
Please take a look